### PR TITLE
Di 24773 default page

### PR DESCRIPTION
**Pull request by liorSela**

### CODE REVIEW

File: shared/metadata.ts [hunk 1]
```diff
@@ -1,5 +1,57 @@
-import { FormDataView } from "@pepperi-addons/papi-sdk";
-import { UsersLists } from "./entities";
+import { AddonDataScheme, FormDataView } from "@pepperi-addons/papi-sdk";
+import { USERS_LISTS_TABLE_NAME, UsersLists } from "./entities";
+// import {v4 as uuid} from 'uuid';
+
+export const notificationsSlug = {
```
Comment (simcha-waintrob) [inline shared/metadata.ts:5]: NotificationsSlug

File: server-side/notifications.service.ts [hunk 2]
```diff
@@ -979,6 +978,37 @@ class NotificationsService {
         const ans = await Promise.all(ansArray);
         return ans;
     }
+
+    async createNotificationsSlug(){
+        try {
+            const slugBody = notificationsSlug
```
Comment (simcha-waintrob) [inline server-side/notifications.service.ts:984]: comment create if not exist

Comment (simcha-waintrob) [reply server-side/notifications.service.ts:984]: set as variable - const
`/addons/api/4ba5d6f9-6642-4817-af67-c79b68c96977/api/slugs

File: server-side/notifications.service.ts [hunk 3]
```diff
@@ -979,6 +978,37 @@ class NotificationsService {
...
+        catch (error){
+            return {success: false, errorMessage: error}
+        }
+
+    }
+
+    async createNotificationsPage(){
+        try {
+            const pages = await this.papiClient.pages.iter().toArray()
+            const pageBody = notificationsPage
```
Comment (simcha-waintrob) [inline server-side/notifications.service.ts:1000]: NotificationsPage

Comment (simcha-waintrob) [reply server-side/notifications.service.ts:1000]: rename to DefaultNotificationsPage
DefaultNotificationsSlug

File: shared/metadata.ts [hunk 4]
```diff
@@ -1,5 +1,57 @@
...
+    Name: "Notifications",
+    Slug: "notifications",
+    Description: "Notifications Default Slug",
+  }
+}
+
+export const notificationsPage: any = {
+  Description: "Notifications Default Page",
+  Blocks:[{
+    Key:"notifications_95025423-9096-4a4f-a8cd-d0a17548e42e_BlockPage",
```
Comment (simcha-waintrob) [inline shared/metadata.ts:16]: 95025423-9096-4a4f-a8cd-d0a17548e42e replace with config.AddonUUID
and in all occurs

File: shared/metadata.ts [hunk 5]
```diff
@@ -1,5 +1,57 @@
...
+    Relation: {
```
Comment (simcha-waintrob) [inline shared/metadata.ts:17]: try to remove Relation - in the new version

File: server-side/installation.ts [hunk 6]
```diff
@@ -109,55 +114,65 @@ export async function upgrade(client: Client, request: Request): Promise<any> {
...
         const userDeviceResourceRes = await createUserDeviceResource(papiClient);
-        return { success: notificationsUsersListsRes.success && 
-            migrateUpLogRes.success && 
-            defaultListRes.success && 
-            notificationsLogViewRes.success && 
-            relationsRes.success && 
-            settingsRelationsRes.success && 
-            userDeviceResourceRes.success, 
+
+        const slugRes = await service.createNotificationsSlug();
```
Comment (simcha-waintrob) [inline server-side/installation.ts:118]: wrap both line in one function

File: server-side/notifications.service.ts [hunk 7]
```diff
@@ -979,6 +978,40 @@ class NotificationsService {
...
+            const slugsUrl = `/addons/api/4ba5d6f9-6642-4817-af67-c79b68c96977/api/slugs`
+            const slugBody = DefaultNotificationsSlug
+            const existingSlugs = await this.papiClient.get(slugsUrl)
+            if(!existingSlugs.find(slug => slug.Slug === slugBody.slug.Slug)){
+                await this.papiClient.post(slugsUrl, slugBody)
+            }
+            return {success: true, errorMessage: ''}
+        }
+        catch (error){
+            return {success: false, errorMessage: error}
```
Comment (simcha-waintrob) [inline server-side/notifications.service.ts:994]: don't fail the installation / upgrade if you can't create the default notifications page & slug

File: server-side/notifications.service.ts [hunk 8]
```diff
@@ -979,6 +978,40 @@ class NotificationsService {
...
+        }
+
+    }
+
+    // create notifications page if not exists already
+    async createNotificationsPage(){
+        try {
+            const pages = await this.papiClient.pages.iter().toArray()
+            const pageBody = DefaultNotificationsPage
+            if(!pages.find(page => page.Blocks[0].Key === pageBody.Blocks[0].Key)){
```
Comment (simcha-waintrob) [inline server-side/notifications.service.ts:1004]: a manually added page have the same key as this?
if no, this may create ad additional page?

Comment (liorSela) [reply server-side/notifications.service.ts:1004]: in our code the page is being created with specific key so it wouldnt be created twice, if a user is creating additional page i cant do anything about it

**Review by simcha-waintrob — APPROVED** [review]

**Pull request merged**